### PR TITLE
Added better build process

### PR DESCRIPTION
**Pull request by samturrell**
Removed module specific code from the source and added it to a package wrapper instead. This allows exporting for many different bundler types, and unpolluted logic.

Also added auto updating public version helper property.

This build process is used on some of my other modules and seems to work well. Let me know your thoughts @limonte.

### CODE REVIEW

**Comment by limonte** [comment]
Thank you @samturrell for this contribution! Looks good, but this PR is pretty big, in future please make smaller PRs :) Give me some time to test your code and understand it.

**Comment by limonte** [comment]
@Tonkpils your opinoin on this PR will be very useful.

File: src/sweetalert2.js [hunk 1]
```diff
@@ -1,1253 +1,1232 @@
...
-      };
-
-      sweetAlert.enableButtons();
-      sweetAlert.disableLoading();
-      sweetAlert.resetValidationError();
-
-      window.onfocus = function() {
-        // When the user has focused away and focused back from the whole window.
-        window.setTimeout(function() {
-          // Put in a timeout to jump out of the event sequenc
```
Comment (limonte) [inline src/sweetalert2.js]: When I'm adding new features of fixing bugs, I'm using [index.html](https://github.com/limonte/sweetalert2/blob/master/index.html) to test swal2.

For dev env I'm changing [line 18](https://github.com/limonte/sweetalert2/blob/master/index.html#L18) to `src/sweetalert2.js` so I can easy debug unminified source code.

With this change I'm getting `sweetalert2.js:1232 Uncaught SyntaxError: Unexpected token export`. I want to be able to run and test index.html with unminified version.

Comment (limonte) [inline src/sweetalert2.js]: When I'm adding new features of fixing bugs, I'm using [index.html](https://github.com/limonte/sweetalert2/blob/master/index.html) to test swal2.

For dev env I'm changing [line 18](https://github.com/limonte/sweetalert2/blob/master/index.html#L18) to `src/sweetalert2.js` so I can easy debug unminified source code.

With this change I'm getting `sweetalert2.js:1232 Uncaught SyntaxError: Unexpected token export`. I want to be able to run and test index.html with unminified version.

**Comment by limonte** [comment]
I have no idea why Github marks my comment above as "commented on an outdated diff" :)

File: src/sweetalert2.js (hunk 1 above)
Comment (limonte) [inline src/sweetalert2.js]: ignore my comment above, I see now `dist/sweetalert.js` :)

**Comment by limonte** [comment]
Great job @samturrell! :+1: :+1: :+1: 

I'm ready to merge this PR after green light from @Tonkpils as he is the author of module specific code :)

**Comment by samturrell** [comment]
Yeah, Github doesn't perform very well with large PR's.

PR is only big because of the removal of the module closure and newly compiled assets. Not sure how I can make it any smaller :)

RE: export token error - I'd probably avoid testing using this file, if only for the reason that it's not what any end user should be using.

**Comment by samturrell** [comment]
Awesome :)

**Comment by limonte** [comment]
@Tonkpils let us now if you have any comment on this PR, thanks!

**Comment by limonte** [comment]
@samturrell https://github.com/limonte/sweetalert2/releases/tag/v3.1.0

**Comment by limonte** [comment]
@samturrell I'm a little bit late with this question but can you please explane why do we need `dist/sweetalert2.min.js.gz`?

**Comment by samturrell** [comment]
@limonte Just as an alternative means of downloading the plugin, less bandwidth with a compressed package.

**Comment by Tonkpils** [comment]
This all looked good to me, I'll test it out with my ember addon 👍

**Pull request merged**